linkis-engineconn-plugins/openlookeng/src/main/java/org/apache/linkis/engineplugin/openlookeng/OpenLooKengECPlugin.java

```diff
@@ -34,8 +34,8 @@
 
 public class OpenLooKengECPlugin implements EngineConnPlugin {
 
-  private final Object resourceLocker = new Object();
-  private final Object engineFactoryLocker = new Object();
+  private volatile Object resourceLocker = new Object();
+  private volatile Object engineFactoryLocker = new Object();
 
   private EngineResourceFactory engineResourceFactory;
   private EngineConnFactory engineFactory;
```

Comment:
the `volatile` key world is for the EngineResourceFactory and EngineConnFactory not the lock object.

Reply:
ok,I change the code